### ISSUE TITLE
Hogwarts plugin

### ISSUE THREAD

**Issue by kiliankoe** [fsr/matthias#29]
> 10 points to @matthias

Record and store points for members. Not sure how to persist this though outside of memory or redis. A file would be thrown away by dokku on deploy.

Idea via [Dumblescore](http://wobbly.space/post/139137533296/dumblescore-a-face-for-a-slack-bot-that-allows)

**Comment by kiliankoe** [comment]
Maybe not as a matthias plugin but just a webhook and stored elsewhere?

**Comment by kiliankoe** [comment]
Since matthias is no longer being deployed via dokku, persistence shouldn't be an issue anymore. Just throw everything into a json file \o/

**Comment by kiliankoe** [comment]
Or use something like this -> https://github.com/eliawry/hogwarts

**Comment by kiliankoe** [comment]
As discussed: It would be awesome to have four houses named after four equally bad programming languages so that no one can play favorites. Suggestions would be PHP, Basic, awk and bash 🙃 A sorting hat throws you in a random house that you then learn to love and earn points for with your fellow housemates \o/

**Comment by JustusAdam** [comment]
One thing though: In the real Hogwarts you can choose. I'd much rather have it so that we do have actual good houses and you can choose one, but there's only a limited number and they should be distinctive so that the choice is meaningful. Aka don't make the houses "C, C#, Objective-C, and C++" but something like one from each of the following:

| Lang set | distinct feature(s) |
| --- | --- |
| C, Rust, C++, ASM | close to hardware and fast |
| Python, BASIC, Ruby, JavaScript | "comfortable" scripting languages (glue code) |
| Swift, Java, C# | OO OO OOOOOO OOO OO (& Mainstream) |
| Haskell, F#, Clojure, Elm, OCaml | Functional |
| Brainfuck, Makefile, AWK, SQL, XML | weird |

**Comment by sjmielke** [comment]
The problem with serious categories? :pitchfork:.

**Comment by kiliankoe** [comment]
> In the real Hogwarts you can choose

You can? I thought that's what the hat is for. "Real" Hogwarts btw 😄

> Swift, Java = OO & Mainstream

Well fu for putting Swift and Java on the same pile! :D

**Comment by JustusAdam** [comment]
> > In the real Hogwarts you can choose
> 
> You can? I thought that's what the hat is for. "Real" Hogwarts btw 😄

Yes you can, have you not read the books? (or watched the movie?)

**Comment by kiliankoe** [comment]
Both, but it's been a few years.

**Comment by sjmielke** [comment]
> > > In the real Hogwarts you can choose
> > 
> > You can? I thought that's what the hat is for. "Real" Hogwarts btw :smile:
> 
> Yes you can, have you not read the books? (or watched the movie?)

But only in some cases and if you have really, really strong prefences and who has thes... wait, we're talking about programming languages...

**Comment by JustusAdam** [comment]
> Well fu to for putting Swift and Java on the same pile! :D

Which pile would you like it on:

| lang set | feature |
| --- | --- |
| Java, C#, Swift | really honestly truly fully platform independent |
| JavaScript, Swift | WRITE YOUR FIRST REALLY ORIGINAL APP TODAY IN 4 EASY STEPS |

**Comment by JustusAdam** [comment]
>  "Real" Hogwarts btw :smile:

What are you trying to say? I am still waiting for the letter.
